qemu-check.exp

```diff
@@ -179,6 +185,10 @@ if {$::env(XEN_BOOT) == "y"} {
 }
 if {$tests == "all" || $tests == "trusted-keys"} {
 	# Invoke Trusted Keys tests
-	set basedir [file dirname $argv0]
 	source $basedir/trusted-keys.exp
 }
+if {$tests == "all" || $tests == "rust"} {
+	if {$::rust_enable==0} { return }
```

Comment:
Prefer with spaces around ==

Reply:
Also I would prefer adding the test to the above check so that is it easier to extend this script:
```
	if {{$tests == "all" || $tests == "rust"} && $::rust_enable == 1} {
		...
```

Reply:
I have tried it before but looks like expect script doesn't like double braces as follows:

```
expected boolean value but got "$tests == "all" || $tests == "rust""
    while executing
"if {{$tests == "all" || $tests == "rust"} && $::rust_enable == 1} {
	# Invoke Rust tests
	source $basedir/rust.exp
}"
    (file "/home/sumit/build/upstream/optee_br/build/../build/qemu-check.exp" line 190)
make: *** [Makefile:543: check] Error 1

```

Reply:
> I have tried it before but looks like expect script doesn't like double braces as follows:
> 
> ```
> expected boolean value but got "$tests == "all" || $tests == "rust""
>     while executing
> "if {{$tests == "all" || $tests == "rust"} && $::rust_enable == 1} {
> 	# Invoke Rust tests
> 	source $basedir/rust.exp
> }"
>     (file "/home/sumit/build/upstream/optee_br/build/../build/qemu-check.exp" line 190)
> make: *** [Makefile:543: check] Error 1
> ```

In fact parentheses should be used, not braces. Sorry for the misleading comment.
```
if {($tests == "all" || $tests == "rust") && $::rust_enable == 1} {
```
See https://stackoverflow.com/questions/11260659/the-if-statement-in-tcl.

Reply:
Yeah that works, thanks for letting me know.